src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -487,49 +480,61 @@ private AIState requestMaterials()
             job.complete();
         }
 
-        //todo as soon as material handling has been implemented this should be set to work!
         //We need to deal with materials
-        if (!Configurations.builderInfiniteResources)
+        if (!Configurations.builderInfiniteResources && !job.getWorkOrder().isRequested())
         {
-            //TODO thread this
             while (job.getStructure().findNextBlock())
             {
                 if (job.getStructure().doesStructureBlockEqualWorldBlock())
                 {
                     continue;
                 }
 
-                @Nullable Block block = job.getStructure().getBlock();
-                @NotNull ItemStack itemstack = new ItemStack(block, 1);
+                @Nullable final Block block = job.getStructure().getBlock();
 
-                Block worldBlock = BlockPosUtil.getBlock(world, job.getStructure().getBlockPosition());
+                @NotNull final ItemStack itemstack = new ItemStack(block, 1);
 
-                if (itemstack.getItem() != null
-                      && block != null
+                final Block worldBlock = BlockPosUtil.getBlock(world, job.getStructure().getBlockPosition());
+
+                if (block != null
                       && block != Blocks.AIR
                       && worldBlock != Blocks.BEDROCK
                       && !(worldBlock instanceof AbstractBlockHut)
-                      && !isBlockFree(block, 0)
-                      && checkOrRequestItems(new ItemStack(block)))
+                      && !isBlockFree(block, 0))
                 {
-                    job.getStructure().reset();
-                    return this.getState();
+                    @NotNull final IBlockState blockState = job.getStructure().getBlockState();
+                    if(blockState instanceof BlockBed && blockState.getValue(BlockBed.PART).equals(BlockBed.EnumPartType.FOOT))
+                    {
+                        continue;
+                    }
+                    else if(blockState instanceof BlockDoor && blockState.getValue(BlockDoor.HALF).equals(BlockDoor.EnumDoorHalf.LOWER))
+                    {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 507. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1871)